### ISSUE TITLE
False negative ``reimported`` when using an import alias

### ISSUE THREAD

**Issue by ddangu525** [PyCQA/pylint#4836]
### Current problem

I think it would be nice if Pylint could check if there is same import name from different packages.
For example, in the code below, import order is fine, but the 'Path' imports overlap, causing errors somewhere.
```python
from pathlib import Path
from FastAPI import Path
```

### Desired solution

overlapping imports found. `Path`

**Comment by Pierre-Sassoulas** [comment]
Thank you for your proposition, it makes sense.

**Comment by DudeNr33** [comment]
The given example already triggers the message ``reimported``: ``Reimport 'Path' (imported line 8)``.
What is not covered yet are aliased imports, e.g.:

```python
from pathlib import Path
from some_other_lib import CustomPath as Path
```

or

```python
from pathlib import Path
import FastAPI.Path as Path
```

So probably it would be best to improve the existing check to also cover aliased imports.

**Comment by DanielNoord** [comment]
I think we should just adjust the docstring and description of the message. I think its intent is pretty clear and also follows what I would expect it to do.

That makes this a double issue:
1) Fix docstring and description of the message
2) Fix the false negative

**Comment by Pierre-Sassoulas** [comment]
Should we also rename the message from ``reimported`` to ``reimported-or-shadowed`` ?

**Comment by DanielNoord** [comment]
I think that creates unnecessary churn and `old-message` stuff. In the end the offending line still contains the word `import` so it is clear that the message is related to "importing and everything related to it".

**Comment by clavedeluna** [comment]
I'll work on this

**Comment by Pierre-Sassoulas** [comment]
@DanielNoord I think ``reimported`` is different than something like ``import-name-clash``or ``reimport-shadowing", "reimported" feel like what should be raised only for:
```pyhon
from pathlib import Path
from pathlib import Path
```
Which is a lot less problematic than shadowing another import like in:
```python
from pathlib import Path
from FastAPI import Path
```
I think we should separate the two use case, with a different message for each, what do you think ?

**Comment by DanielNoord** [comment]
I'm not sure. Why would you ever want to enable one of those but not the other? They both warn of significant code smells that seem very related.

I'm not against it, but just feel like unnecessary work with minimal benefit. We can focus on many other issues.

**Comment by Pierre-Sassoulas** [comment]
Let's separate the two checks between ``reimported`` and ``shadowed-import`` like @DudeNr33 suggested in its (original message)[https://github.com/PyCQA/pylint/issues/4836#issuecomment-907793766]. Unclear message and checker doing multiple things means user have a hard time understanding and we change the check more and create more doc to accommodate the multiple things it needs to do or explain. This is more work in the end.

**Comment by LLyaudet** [comment]
Hello,
I just had a similar case with:
```python3
from some_module import something

...
def bla():
    def blabla():
        from some_module import something as something_
        ...
    # use blabla
```
In some very rare cases, it may make sense to add another alias in this kind of generated functions.
But after thinking about it first and reading this issue,
I think it is more convenient to have the warning,
and add a pylint disable comment when this is really necessary.

Best regards,
     Laurent Lyaudet
